lldb/tools/lldb-dap/Handler/RequestHandler.cpp

```diff
@@ -225,4 +225,11 @@ void RequestHandler::PrintWelcomeMessage() {
 #endif
 }
 
+bool RequestHandler::HasInstructionGranularity(
+    const llvm::json::Object &request) {
+  if (std::optional<llvm::StringRef> value = request.getString("granularity"))
+    return value == "instruction";
+  return false;
+}
```

Comment:
Can we change the name of the parameter to `arguments`?

We shouldn't check the request itself, but the arguments, e.g. the request would be formatted like:
```json
{
  "type": "request",
  "command": "next",
  "seq": 1,
  "arguments": {
    "granularity": "instruction"
  }
}
```